buildkite/src/Command/MinaArtifact.dhall

```diff
@@ -20,6 +20,7 @@ let dirtyWhen = [
   S.strictlyStart (S.contains "buildkite/src/Command/MinaArtifact"),
   S.exactly "buildkite/scripts/build-artifact" "sh",
   S.exactly "buildkite/scripts/connect-to-mainnet-on-compatible" "sh",
+  S.strictlyStart (S.contains "buildkite/src/Jobs/Test/TestnetIntegrationTest"),
```

Comment:
I think "buildkite/src/Jobs/Test" is probably a better filter here as most testing jobs depend on these artifacts (or individually list TestnetIntegrationTests, ReplayerTest, DeployTestnetAndDestroy, MigratedIntegrationTests, ConnectToTestnet). Similarly, "buildkite/src/Command" does not depend on these jobs but has the potential to break them, so they should be dirty.

Reply:
more correctly long term the answer is probably more folders for these jobs to better reflect dependency, like if these were all under "buildkite/src/Jobs/Test/Integration/" to separate from unit tests / static analysis.